### PR TITLE
Store Stats

### PR DESCRIPTION
**Pull request by deanshub**
- lint fix
- Storing stats in DB

### CODE REVIEW

File: packages/web/src/app/api/stats/route.ts [hunk 1]
```diff
@@ -1,47 +1,161 @@
...
+import { getServerSession } from 'next-auth';
+import { nextAuthOptions } from '@/utils/auth';
+import { createUser } from '@/db/helpers/createUser';
+import { createAppUserConnection } from '@/db/helpers/createAppUserConnection';
+import { getNullAsUndefined } from '@/utils/getNullAsUndefined';
+import type { PostStatsResponse } from 'in-it-shared-types';
 
 export async function POST(request: Request) {
+  const session = await getServerSession(nextAuthOptions);
+  // TODO: user user session after Amit's PR
```
Comment (deanshub) [inline packages/web/src/app/api/stats/route.ts:19]: todo after #10

File: packages/web/src/app/api/stats/route.ts [hunk 2]
```diff
@@ -1,47 +1,161 @@
...
+  const branch = getNullAsUndefined(form.get('branch') as null | string);
+  const generatingTool = getNullAsUndefined(form.get('generatingTool') as null | string);
+  const generatingToolVersion = getNullAsUndefined(
+    form.get('generatingToolVersion') as null | string,
+  );
+  const repository = getNullAsUndefined(form.get('repository') as null | string);
+  const packagePath = getNullAsUndefined(form.get('packagePath') as null | string);
+  const packageName = getNullAsUndefined(form.get('packageName') as null | string);
+
+  // TODO: read sizes from the file
```
Comment (deanshub) [inline packages/web/src/app/api/stats/route.ts]: TODO

File: packages/web/src/app/api/stats/route.ts [hunk 3]
```diff
@@ -1,47 +1,161 @@
...
+  // if connection between user and app does not exist, create one (no await intentionally)
+  if (user) {
+    createAppUserConnection({
+      appId,
+      userId: user._id,
+    });
+  }
+
+  // TODO: check what's the best unique path for the stats
+  const statsFilePath = `${envirmonet}/${compilation}/stats.json`;
```
Comment (deanshub) [inline packages/web/src/app/api/stats/route.ts]: TODO

File: packages/shared-types/entities.ts [hunk 4]
```diff
@@ -67,3 +67,9 @@ export interface Limit {
     allowedModules?: string[];
   };
 }
+export interface BundleStatsReport {
+  label: string;
```
Comment (shamit6) [inline packages/shared-types/entities.ts:71]: what is it for? you don't use it

File: packages/web/src/app/api/stats/route.ts [hunk 5]
```diff
@@ -1,47 +1,171 @@
...
+      repository,
+      packagePath,
+      name,
+      packageName,
+    });
+    appId = app._id;
+  } else {
+    // update existing app in db (no await intentionally)
+    updateApp(appId, {
+      provider: provider as undefined | 'github' | 'gitlab' | 'bitbucket',
```
Comment (shamit6) [inline packages/web/src/app/api/stats/route.ts]: as undefined | SourceCodeProvider

File: packages/shared-types/entities.ts (hunk 4 above)
Comment (deanshub) [reply packages/shared-types/entities.ts:71]: that's the type for the json that the bundle-analyzer outputs (but just a little bit of it that we actually need)

File: packages/web/src/utils/getUserId.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+export async function getUserId({
+  githubUsername,
+  gitlabUsername,
+  bitbucketUsername,
+  email,
+}: Partial<UserDocument>): Promise<UserDocument | null> {
+  if (githubUsername || gitlabUsername || bitbucketUsername) {
+    // query db for userId using userNameInProvider, provider
+    const users = await User.find({
```
Comment (shamit6) [inline packages/web/src/utils/getUserId.ts:12]: not sure but maybe you can send the provider and use this function instead 
https://github.com/deanshub/in-it/blob/3061d9f04e319204e2dcbd22e368d10d70a8d1b8/packages/web/src/db/queries.ts#L9-L16

```typescript

```

File: packages/web/src/db/models/Stats.ts [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+  repository: {
+    type: String,
+    // required: true,
+    index: true,
+  },
+  packagePath: {
+    type: String,
+    // required: true,
+    index: true,
```
Comment (shamit6) [inline packages/web/src/db/models/Stats.ts]: you don't need index on this field you need a combined index on repository, packagePath fields

**Review by shamit6 — APPROVED** [review]

**Pull request merged**